spec/formatting.md

```diff
@@ -1,24 +1,113 @@
-# WIP DRAFT MessageFormat 2.0 Formatting Behaviour
+# DRAFT MessageFormat 2.0 Formatting Behaviour
 
 ## Introduction
 
 This document defines the behaviour of a MessageFormat 2.0 implementation
 when formatting a message for display in a user interface, or for some later processing.
 
-The document is part of the MessageFormat 2.0 specification,
-the successor to ICU MessageFormat, henceforth called ICU MessageFormat 1.0.
+Message formatting is defined in terms of the following stages:
 
-## Literal Resolution
+- **_Resolution_** determines the value of a part of the message,
+  such as an _expression_, with reference to the current _formatting context_.
+  This may include multiple steps,
+  such as looking up the value of a variable and calling formatting functions.
+  The resolved value is not necessarily in the shape it will finally take,
+  but is "formattable", i.e. it contains everything required by the eventual formatting.
 
-The resolved value of _text_, _literal_ and _nmtoken_ tokens
+  > For example, the resolved value of the _expression_ `{|0.40| :number style=percent}`
+  > could be an object such as
+  >
+  > ```
+  > { value: Number('0.40'),
+  >   formatter: NumberFormat(locale, { style: 'percent' }) }
+  > ```
+  >
+  > The shape of this value is completely implementation-dependent,
+  > and different implementations may choose to perform different levels of resolution.
```

Comment:
here `MAY` is normative and should be formatted normatively

Reply:
Hmmm. This should probably be reformulated a bit differently, as it's currently in an example's `>` block which ought to mean it's non-normative.